src/main/java/org/eclipse/yasson/internal/ClassParser.java

```diff
@@ -173,14 +174,33 @@ private void parseMethods(Class<?> clazz, JsonbAnnotatedElement<Class<?>> classE
         for (Method method : declaredMethods) {
             String name = method.getName();
             //isBridge method filters out methods inherited from interfaces
-            if (!isPropertyMethod(method) || method.isBridge()) {
+            if (!isPropertyMethod(method) || method.isBridge() || isSpecialCaseMethod(clazz, method)) {
                 continue;
             }
             final String propertyName = toPropertyMethod(name);
 
             Property property = registerMethod(propertyName, method, classElement, classProperties);
         }
     }
+    
+    /**
+     * Filter out certain methods that get forcibly added to some classes.
+     * For example the public groovy.lang.MetaClass X.getMetaClass() method from Groovy classes
+     */
+    private boolean isSpecialCaseMethod(Class<?> clazz, Method m) {
```

Comment:
Copyright year should be 2019 instead of 2018.

Reply:
@Verdent fixed -- ready for re-review

Reply:
actually, just going to merge since I assume you have no further review comments